### PR TITLE
Demonstrate passing through arguments with z-forward test function

### PR DESCRIPTION
**Pull request by ras0219-msft**
```
$ vcpkg z-forward --a --b --c --vcpkg-root=/workspaces/vcpkg --debug
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] BuiltinRegistry initialized with: ""
[DEBUG] Using vcpkg-root: /workspaces/vcpkg
[DEBUG] Using installed-root: /workspaces/vcpkg/installed
[DEBUG] BuiltinRegistry initialized with: ""
[DEBUG] Using buildtrees-root: /workspaces/vcpkg/buildtrees
[DEBUG] Using downloads-root: /workspaces/vcpkg/downloads
[DEBUG] Using packages-root: /workspaces/vcpkg/packages
[DEBUG] Using scripts-root: /workspaces/vcpkg/scripts
[DEBUG] Using ports-root: /workspaces/vcpkg/ports
[DEBUG] Using versions-root: /workspaces/vcpkg/versions
[DEBUG] system(insert-executable-name-here -- z-forward --a --b --c --vcpkg-root=/workspaces/vcpkg --debug)
sh: 1: insert-executable-name-here: not found
[DEBUG] system() returned 32512 after 559 us
[DEBUG] /workspaces/vcpkg-tool/src/vcpkg/commands.zforward.cpp(14)
[DEBUG] Exiting after %s (%d us)
```

Additional parameters can be added before the `--` to easily distinguish them from user-provided options.

### CODE REVIEW

**Comment by BillyONeal** [comment]
Thanks for the demonstration! Looks good to me.

File: src/vcpkg/commands.cpp [hunk 1]
```diff
@@ -111,6 +113,7 @@ namespace vcpkg::Commands
             {"format-manifest", &format_manifest},
             {"x-ci-verify-versions", &ci_verify_versions},
             {"x-add-version", &add_version},
+            {"x-forward", &test_forward},
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.cpp:116]: shouldn't this be `z-forward`, since it's internal?

File: src/vcpkg/commands.x-forward.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include <vcpkg/base/system.process.h>
+
+#include <vcpkg/commands.x-forward.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+
+namespace vcpkg::Commands::X_Forward
+{
+    void perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths&)
+    {
+        Command cmd("vcpkg-utility-impl");
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.x-forward.cpp:10]: where does `vcpkg-utility-impl` come from?

Comment (ras0219-msft) [reply src/vcpkg/commands.x-forward.cpp:10]: It comes from nowhere -- it's a demonstrative "insert executable name here".

**Comment by ras0219-msft** [comment]
Ready to merge by any approver

**Comment by BillyONeal** [comment]
> Ready to merge by any approver -- I plan to leave this demonstration command in the codebase for now

I think it makes sense to just leave this in PR form and the effects just land in the first command that actually does this

**Pull request closed without merging**